addon/components/au-file-upload.js

```diff
@@ -67,43 +69,53 @@ export default class AuFileUploadComponent extends Component {
       const response = yield file.upload(this.endPoint, {
         'Content-Type': 'multipart/form-data',
       });
-      const fileId = response.body?.data?.id;
+      const body = yield response.json();
+      const fileId = body?.data?.id;
       return fileId;
     } catch (e) {
-      this.uploadErrorData = [...this.uploadErrorData, { filename: file.name }];
+      this.addError(file);
       this.removeFileFromQueue(file);
       return null;
     }
   }
 
-  hasValidationErrors(file) {
-    // TODO: see if we can split the side-effects from the
-    // non-side-effects.  The name doesn't suggest this will yield a
-    // state change.
-    if (file.size > this.maxFileSizeMB * Math.pow(1024, 2)) {
-      this.uploadErrorData = [
-        ...this.uploadErrorData,
-        {
-          filename: file.name,
-          error: `Bestand is te groot (max ${this.maxFileSizeMB} MB)`,
-        },
-      ];
-      this.removeFileFromQueue(file);
-      return true;
+  @action
+  filter(file, files, index) {
+    let isFirstFile = index === 0;
+
+    if (isFirstFile) {
+      this.resetErrors();
+    } else {
+      if (!this.args.multiple) {
+        // We only upload the first file if `@multiple` is not set to true. This matches the behavior of ember-file-upload v4.
+        return false;
+      }
     }
-    return false;
-  }
 
-  @action
-  resetErrors() {
-    this.uploadErrorData = [];
+    let hasValidFileSize = file.size < this.maxFileSizeMB * Math.pow(1024, 2);
+    if (!hasValidFileSize) {
+      this.addError(file, `Bestand is te groot (max ${this.maxFileSizeMB} MB)`);
+      return false;
+    }
+
+    if (typeof this.args.filter === 'function') {
+      let shouldBeUploaded = this.args.filter(...arguments);
+
+      if (shouldBeUploaded) {
+        return true;
+      } else {
+        this.addError(file, this.helpTextFileNotSupported);
+        return false;
+      }
+    } else {
+      return true;
```

Comment:
At the moment, if no `@filter` argument is provided the file can be uploaded. This means that `@accept` will only be used by the file input to soft disable some file types in the file selector popup.

We could implement some logic to do basic file filtering based on the extension or mime type, similar to how it worked in ember-file-upload v4.

Reply:
As discussed in Rocket.chat, making this as frictionless as possible for projects (and end-users) is preferred, so we should try to add a basic mime-type / extension based validation system similar to how it worked in ember-file-upload v4.